data-prepper-plugins/dynamodb-source/src/main/java/org/opensearch/dataprepper/plugins/source/dynamodb/DynamoDBSource.java

```diff
@@ -0,0 +1,89 @@
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.model.source.Source;
+import org.opensearch.dataprepper.model.source.SourceCoordinationStore;
+import org.opensearch.dataprepper.plugins.source.dynamodb.coordination.Coordinator;
+import org.opensearch.dataprepper.plugins.source.dynamodb.coordination.DefaultCoordinator;
+import org.opensearch.dataprepper.plugins.source.dynamodb.coordination.PartitionFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@DataPrepperPlugin(name = "dynamodb", pluginType = Source.class, pluginConfigurationType = DynamoDBSourceConfig.class)
+public class DynamoDBSource implements Source<Record<Event>> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DynamoDBSource.class);
+
+    private static final String SOURCE_COORDINATOR_METRIC_PREFIX = "source-coordinator";
+
+    private final PluginMetrics pluginMetrics;
+
+    private final DynamoDBSourceConfig sourceConfig;
+
+    private final PluginFactory pluginFactory;
+
+    private final SourceCoordinationStore coordinationStore;
+
+    private final Coordinator coordinator;
+
+    private final DynamoDBService dynamoDBService;
+
+
+    @DataPrepperPluginConstructor
+    public DynamoDBSource(PluginMetrics pluginMetrics, final DynamoDBSourceConfig sourceConfig, final PluginFactory pluginFactory, final PluginSetting pluginSetting, final AwsCredentialsSupplier awsCredentialsSupplier) {
+        LOG.info("Create DynamoDB Source");
+        this.pluginMetrics = pluginMetrics;
+        this.sourceConfig = sourceConfig;
+        this.pluginFactory = pluginFactory;
+
+
+        // Load Coordination Store vai PluginFactory
+        PluginSetting sourceCoordinationStoreSetting = new PluginSetting(sourceConfig.getCoordinationStoreConfig().getPluginName(), sourceConfig.getCoordinationStoreConfig().getPluginSettings());
+        sourceCoordinationStoreSetting.setPipelineName(SOURCE_COORDINATOR_METRIC_PREFIX);
+        coordinationStore = pluginFactory.loadPlugin(SourceCoordinationStore.class, sourceCoordinationStoreSetting);
+        String pipelineName = pluginSetting.getPipelineName();
+
+        // Create and initialize coordinator
+        coordinator = new DefaultCoordinator(coordinationStore, pipelineName, new PartitionFactory());
+        coordinator.initialize();
+
+        ClientFactory clientFactory = new ClientFactory(awsCredentialsSupplier, sourceConfig.getAwsAuthenticationConfig());
+
+        // Create DynamoDB Service
+        dynamoDBService = new DynamoDBService(coordinator, clientFactory, sourceConfig, pluginMetrics);
+        dynamoDBService.init();
+    }
+
+    @Override
+    public void start(Buffer<Record<Event>> buffer) {
+        LOG.info("Start Processing");
+        dynamoDBService.start(buffer);
+    }
+
+
+    @Override
+    public void stop() {
+        LOG.info("Stop DynamoDB Source");
+        dynamoDBService.shutdown();
```

Comment:
Should wrap this call in an `if (Objects.nonNull(dynamoDBService)` check. Not a big deal but NullPointer would happen if this gets run before line 66

Reply:
Will udpate.